src/PREfastXmlSarifConverter/PREfastXmlSarifConverter.cpp

```diff
@@ -161,6 +258,30 @@ class XmlToSarifConverter
 					pReader->GetValue(&wszValue, &cchValue);
 					defect.SetFunctionLine(wszValue);
 				}
+				else if (wcscmp(wszLocalName, L"PROBABILITY") == 0)
+				{
+					HRESULT hr = pReader->Read(&nodeType);
+					pReader->GetValue(&wszValue, &cchValue);
+					defect.SetProbability(wszValue);
+				}
+				else if (wcscmp(wszLocalName, L"RANK") == 0)
+				{
+					HRESULT hr = pReader->Read(&nodeType);
+					pReader->GetValue(&wszValue, &cchValue);
+					defect.SetRank(wszValue);
+				}
+				else if (wcscmp(wszLocalName, L"CATEGORY") == 0)
+				{
+					HRESULT hr = pReader->Read(&nodeType);
+					pReader->GetValue(&wszValue, &cchValue);
+					ReadCategory(pReader, defect.m_category); 
```

Comment:
You use the method "`ReadCategory`" to read both the `CATEGORY` and the `ADDITIONALINFO`. This method really reads an arbitrary element `<name>value</name>` and adds it to a dictionary which is essentially a property bag, right? Perhaps it should be named `ReadProperty`.